internal/adapter/lsp/util.go

```diff
@@ -56,3 +60,59 @@ func (b *jsonBoolean) UnmarshalJSON(data []byte) error {
 	}
 	return nil
 }
+
+type linkInfo struct {
+    note *core.MinimalNote
+	location *protocol.Location
+	title    *string
+}
+
+func linkNote(notebook *core.Notebook, documents *documentStore, context *glsp.Context, info *linkInfo) (bool, error) {
```

Comment:
You can simplify this a bit by returning a single `error`, as `bool` is `true` only when the error is `nil`.